### PR TITLE
Local invoke AWS CF function resolution for env variable : Ref

### PR DESCRIPTION
**Pull request by fredericbarthelet**
Addresses: #7087 #8147

### CODE REVIEW

**Comment by fredericbarthelet** [comment]
@medikoo I could add an intermediate service responsible for resolving both imports and refs in order to simplify invoke local code and abstracting away the different functions that can be resolved. Would you prefer such service to be implemented or do you like it better this way ?

File: lib/plugins/aws/utils/resolveCfRefValue.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const _ = require('lodash');
+const ServerlessError = require('../../../classes/Error').ServerlessError;
+
+function resolveCfRefValue(provider, resourceLogicalId, sdkParams = {}) {
+  return provider
+    .request(
+      'CloudFormation',
+      'listStackResources',
+      _.merge(sdkParams, { StackName: provider.naming.getStackName() })
```
Comment (medikoo) [inline lib/plugins/aws/utils/resolveCfRefValue.js:11]: Let's use `Objec.assign`

File: lib/plugins/aws/invokeLocal/index.js [hunk 2]
```diff
@@ -200,7 +201,23 @@ class AwsInvokeLocal {
         );
       });
 
-      return BbPromise.all(promises).then(() => {
+      const fnRefPromises = _.entries(configuredEnvVars).map(([name, value]) => {
```
Comment (medikoo) [inline lib/plugins/aws/invokeLocal/index.js:204]: It'll be nicer to use one`map` iteration for both `'Fn::ImportValue'` and `'Ref'`

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @fredericbarthelet it looks good! Still let me know what you think about my suggetsions

> @medikoo I could add an intermediate service responsible for resolving both imports and refs in order to simplify invoke local code and abstracting away the different functions that can be resolved. Would you prefer such service to be implemented or do you like it better this way ?

Do you think, we can (and should) use it then in some other places? If yes, then yes I think it might be nice to abstract it away.

**Comment by fredericbarthelet** [comment]
@medikoo I updated my PR with your comments. I don't see any use yet for a service requiring as well import and ref intrinsic functions. When the case arise, I shall implement such service.

**Review by medikoo — COMMENTED** [review]
Thank you @fredericbarthelet looks great. Still shouldn't we also mention support for `Ref` in docs here: https://github.com/fredericbarthelet/serverless/blob/13e1bf7a7d2aba877a2cbd6e3c0c59689fbe8964/docs/providers/aws/cli-reference/invoke-local.md#local-function-invocation-setting-environment-variables ?

**Comment by fredericbarthelet** [comment]
@medikoo you're right, I updated the documentation as well :)

**Review by medikoo — APPROVED** [review]
Thank you @fredericbarthelet !

**Pull request merged**